tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/Path.java

```diff
@@ -71,10 +71,64 @@ public Path(String device, String measurement) {
     }
     this.device = device;
     this.measurement = measurement;
-    this.fullPath = device + TsFileConstant.PATH_SEPARATOR + measurement;
+    this.fullPath = device + TsFileConstant.PATH_SEPARATOR + (measurement.contains(TsFileConstant.PATH_SEPARATOR) ? "\"" + measurement + "\"": measurement);
+  }
+
+  /**
+   * extract device and measurement info from complete path string
+   *
+   * @param pathSc complete path string
+   * @return str[0] is the device string, and str[1] is the measurement string
+   */
+  private void init(String pathSc) {
+    pathSc = pathSc.trim();
+    int i = pathSc.length() - pathSc.replace("\"", "").length();
+    int j = pathSc.length() - pathSc.replace("\'", "").length();
```

Comment:
You are using expensive operations like `replace()` and `split()`, I would suggest replacing it with a char scan like:
```
    int firstSingleQuotePos = -1;
    int lastSingleQuotePos = -1;
    int firstDoubleQuotePos = -1;
    int lastDoubleQuotePos = -1;
    for (int i = 0; i < pathSc.length(); i++) {
      switch (pathSc.charAt(i)) {
        case '\'':
          if (firstDoubleQuotePos != -1) {
            // throw exception
          }
          if (firstSingleQuotePos == -1) {
            firstSingleQuotePos = i;
          } else if (lastSingleQuotePos == -1) {
            lastSingleQuotePos = i;
          } else {
            // throw exception
          }
          break;
        case '\"':
          // do similar things
      }
    }
    if (firstDoubleQuotePos != -1 && lastDoubleQuotePos == -1 ||
        firstSingleQuotePos != -1 && lastSingleQuotePos == -1) {
      // throw exception
    }
    if (lastDoubleQuotePos != -1 && lastDoubleQuotePos != pathSc.length() - 1 ||
    lastSingleQuotePos != -1 && lastSingleQuotePos != pathSc.length() - 1) {
      // throw exception
    }
    if (lastSingleQuotePos != -1) {
      String device = firstSingleQuotePos > 0 ? pathSc.substring(0, firstSingleQuotePos - 1) : "";
      String measurement = pathSc.substring(firstSingleQuotePos + 1, pathSc.length() - 1);
    } else if (lastDoubleQuotePos != -1) {
      // do similar things
    } else {
      int lastSeparatorPos = pathSc.lastIndexOf('.');
      String device = lastSeparatorPos > 0 ? pathSc.substring(0, lastSeparatorPos) : "";
      String measurement = pathSc.substring(lastSeparatorPos + 1);
    }
```

Reply:
Thanks for your suggestion. I modified the method and counted the number of quote marks by traversing the string. So `replace()` method is avioded.